http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -444,6 +461,22 @@ protected AuthChallenge resolveCode(String code) {
             TokenValidator.VerifiedTokens verifiedTokens = tokenValidator.parseAndVerifyToken(idTokenString, tokenString);
             idToken = verifiedTokens.getIdToken();
             token = verifiedTokens.getAccessToken();
+
+            String stateCookieValue = getCookieValue(deployment.getStateCookieName());
```

Comment:
Won't this get the value of the `OAuth_Token_Request_State` cookie instead of the session random value cookie.